app/gui2/src/composables/nodeCreation.ts

```diff
@@ -168,9 +147,39 @@ export function useNodeCreation(
     }
     const id = asNodeId(rhs.id)
     const rootExpression =
-      documentation != null ? Ast.Documented.new(documentation, assignment) : assignment
-    return { rootExpression, id }
+      options.documentation != null ?
+        Ast.Documented.new(options.documentation, assignment)
+      : assignment
+    return { rootExpression, id, inferredType: inferredPrefix }
   }
 
   return { createNode, createNodes, placeNode }
 }
+
+/** Try to infer binding name from AST. This is used when type information from the engine is not available yet. */
+function inferPrefixFromAst(expr: Ast.Ast): string | undefined {
+  if (expr instanceof Ast.Vector) return 'vector'
+  if (expr instanceof Ast.NumericLiteral) return expr.code().includes('.') ? 'float' : 'integer'
+  if (expr instanceof Ast.TextLiteral) return 'text'
+  if (expr instanceof Ast.OprApp) {
+    if (expr.operator.ok && expr.operator.value.code() === '+') return 'sum'
+    if (expr.operator.ok && expr.operator.value.code() === '-') return 'diff'
+    if (expr.operator.ok && expr.operator.value.code() === '*') return 'prod'
+    if (expr.operator.ok && expr.operator.value.code() === '/') return 'quot'
+  }
```

Comment:
This can be greatly simplified into either a lookup table or a switch statement:
```suggestion
  if (expr instanceof Ast.OprApp && expr.operator.ok) {
   return operatorCodeToName[expr.operator.value.code()] ?? 'operator'
  }
```

```ts 
const operatorCodeToName = {
	'+': 'sum',
	// ...
}
```